api/src/test/java/org/openmrs/PersonNameTest.java

```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
 		pn.setFamilyNameSuffix("jr.");
 		pn.setDegree("3");
 		PersonName.setFormat("");
-		Assert.assertEquals(pn.getFullName(), "Sr. Taylor Bob Mark");
+
+		Assert.assertEquals("Sr. Taylor Bob Mark", pn.getFullName());
+	}
+
+	/**
+	 * @see PersonName(String, String, String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces on each name attribute", method = "PersonName")
+	public void personName_shouldTrimLeadingAndTrailingWhitespacesOnEachNameAttribute() throws Exception {
+		PersonName pn = new PersonName(" David ", " Robert ", " Jones ");
+
+		assertThat(pn.getGivenName(), is("David"));
+		assertThat(pn.getMiddleName(), is("Robert"));
+		assertThat(pn.getFamilyName(), is("Jones"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyName(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyName(String)")
+	public void setFamilyName_shouldTrimLeadingAndTrailingWhitespaces() {
+		PersonName pn = new PersonName();
+		pn.setFamilyName(" van Dyke ");
+
+		assertThat(pn.getFamilyName(), is("van Dyke"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyName2(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyName2(String)")
+	public void setFamilyName2_shouldTrimLeadingAndTrailingWhitespaces() {
+		PersonName pn = new PersonName();
+		pn.setFamilyName2(" von Hohenlohe ");
+
+		assertThat(pn.getFamilyName2(), is("von Hohenlohe"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyNamePrefix(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyNamePrefix(String)")
+	public void setFamilyNamePrefix_shouldTrimLeadingAndTrailingWhitespaces() {
+		PersonName pn = new PersonName();
+		pn.setFamilyNamePrefix(" van der ");
+
+		assertThat(pn.getFamilyNamePrefix(), is("van der"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyNameSuffix(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyNameSuffix(String)")
+	public void setFamilyNameSuffix_shouldTrimLeadingAndTrailingWhitespaces() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l498)